[Editor] Pointer type initialization only when the user changes the mode

legraina opened the pull request:
It clears the pointer type of ``CurrentPointers`` outside of the ``DrawingEditor`` to always keep the same, until the user changes the mode.
One of the goal is to keep the same pointer type in case of a refresh of the page or if the user changes the pdf document.

legraina commented on src/display/editor/tools.js:
```diff
@@ -1796,13 +1877,16 @@ class AnnotationEditorUIManager {
    *   edit mode.
    * @param {boolean} [editComment] - true if the mode change is due to a
    *   comment edit.
+   * @param {boolean} [isFromEvent] - true if the mode change is due to an event
+   * (e.g. toolbar button clicked).
```
Should it be ``isFromUser`` or ``isFromAction`` ?

legraina commented:
@timvandermeij @calixteman Hi, I've merged the new master into this branch. It should be ready to merge, if good for you.

timvandermeij commented on src/display/editor/tools.js:
```diff
@@ -1922,6 +1922,8 @@ class AnnotationEditorUIManager {
   /**
    * Change the editor mode (None, FreeText, Ink, ...)
    * @param {number} mode
+   * @param {boolean} [isFromUser] - true if the mode change is due to a
```
Required arguments should be listed before optional ones, so please swap `isFromUser` with `editId` here and in the function definition below.

timvandermeij commented on web/pdf_viewer.js:
```diff
@@ -2500,6 +2500,7 @@ class PDFViewer {
       this.#annotationEditorMode = mode;
       await this.#annotationEditorUIManager.updateMode(
         mode,
+        true,
```
To make it a bit more clear what `true` represents here, given that there is no variable like `isFromKeyboard` that immediately makes the intention clear, we usually add an inline comment like this:

```
await this.#annotationEditorUIManager.updateMode(
  mode,
  editId,
  /* isFromUser = */ true,
  isFromKeyboard,
  ...
);
```

(also here with `editId` before the optional parameters)

legraina replied:
Done

timvandermeij reviewed (commented):
In addition to the comments, please also squash the commits into one, with a good commit message that shortly explains the purpose of the change. Thanks!

legraina committed:
Pointer type initialization only when the user changes the mode

Clear the pointer type of CurrentPointers outside of the DrawingEditor to always keep the same, until the user changes the mode.
One of the goal is to keep the same pointer type in case of a refresh of the page or if the user changes the pdf document.

legraina commented:
The PR is ready.

timvandermeij reviewed (approved):
Looks good to me, with passing integration tests, but I would like a review from @calixteman on this too before merging. Thanks!

legraina commented:
Hi @calixteman, is it possible to merge to try to move to the next PR when you have time ?
Thanks !
